### PR TITLE
Make PyWavelets an optional dependency

### PR DESCRIPTION
**Pull request by lagru**
## Description

As discussed in the meeting today, we might want to make PyWavelets an optional dependency. 
It is not as lightweight (pure Python) as networkx and currently blocks our release. It is also only used for two of our functions: [`denoise_wavelet`](https://scikit-image.org/docs/dev/api/skimage.restoration.html#skimage.restoration.denoise_wavelet) and [`estimate_sigma`](https://scikit-image.org/docs/dev/api/skimage.restoration.html#skimage.restoration.estimate_sigma).

`denoise_wavelet` actually works without PyWavelets for the 1D case if `convert2ycbr` is `True` and `estimate_sigma` is `False`.

I used `xfail` instead of `pytest.importorskip` or `pytest.mark.skip`, so that the above behavior can actually be tested and we notice if the failure is due something different (not an `ImportError`).

I'm marking this as API for lack of a better label. I feel like this would get lost in the maintenance section otherwise. Feel free to change this if you feel like another label fits better.

cc @mkcor, @stefanv, @jarrodmillman 

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
Make PyWavelets an optional dependency which is only required for 
`skimage.restoration.denoise_wavelet` and `skimage.restoration.estimate_sigma`.
```

### CODE REVIEW

File: skimage/restoration/_denoise.py [hunk 1]
```diff
@@ -885,6 +892,8 @@ def denoise_wavelet(image, sigma=None, wavelet='db1', mode='soft',
 
     Examples
     --------
+    >>> import pytest
+    >>> _ = pytest.importorskip('pywt')
```
Comment (lagru) [inline skimage/restoration/_denoise.py:896]: Found this solution here
https://github.com/scikit-image/scikit-image/blob/cb353695fc23eb2a19a58c14d25485e0eeae289b/skimage/feature/_fisher_vector.py#L82-L83 

Would be nice to hide these lines somehow or skip this doctest another way but not sure how.

File: skimage/restoration/j_invariant.py [hunk 2]
```diff
@@ -258,15 +258,14 @@ def calibrate_denoiser(image, denoise_function, denoise_parameters, *,
...
+    >>> from skimage.restoration import denoise_tv_chambolle
     >>> import numpy as np
     >>> img = color.rgb2gray(data.astronaut()[:50, :50])
     >>> rng = np.random.default_rng()
     >>> noisy = img + 0.5 * img.std() * rng.standard_normal(img.shape)
-    >>> parameters = {'sigma': np.arange(0.1, 0.4, 0.02)}
-    >>> denoising_function = calibrate_denoiser(noisy, denoise_wavelet,
+    >>> parameters = {'weight': np.arange(0.01, 0.3, 0.02)}
+    >>> denoising_function = calibrate_denoiser(noisy, denoise_tv_chambolle,
     ...                                         denoise_parameters=parameters)
```
Comment (lagru) [inline skimage/restoration/j_invariant.py:269]: Not sure if replacing `denoise_wavelet` with `denoise_tv_chambolle` here (and elsewhere) makes sense but it prevents having to skip these doctests. Found an example of `denoise_tv_chambolle` being used with `calibrate_denoiser` in [this gallery example](https://scikit-image.org/docs/stable/auto_examples/filters/plot_j_invariant_tutorial.html#comparing-different-classes-of-denoiser).

**Comment by stefanv** [comment]
Is pywt so pervasive in the restoration submodule that the whole submodule has to raise an importerror? Usually we do it per function, but perhaps there aren't enough other functions around to make it worthwhile.

**Comment by lagru** [comment]
> Is pywt so pervasive in the restoration submodule that the whole submodule has to raise an importerror?

Maybe I misunderstand, but the change only raises ImportErrors per function. Though, it's touching more than I originally thought, because `denoise_wavelet` is used in a few additional doc examples and tests.

**Comment by stefanv** [comment]
No worries, I was probably seeing https://github.com/scikit-image/scikit-image/pull/7156/files#diff-f676de72662d8ee6bc9afa9e3ca6733b2c6ee1f23a25dbc192f7aef9712d0591R1020 and thought it was top-level. All good.

**Comment by lagru** [comment]
Finally green. CI really keeping me honest here. :sweat_smile: Learned some new stuff about pytest's xfail API.

**Review by jarrodmillman — APPROVED** [review]
LGTM

**Review by stefanv — APPROVED** [review]

**Pull request merged**